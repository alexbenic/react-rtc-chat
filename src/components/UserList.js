import React from "react";

const UserList = ({ current, users }) => {
  const list = users.filter(user => user.id !== current.id);

  return (
    <div className="flex-auto mt2">
      {list.map(user => <ListItem key={user.id}>{user.nick}</ListItem>)}
    </div>
  );
};

const ListItem = ({ children }) => {
  return (
    <li className="flex items-center mt2 mb2 pb1 pt1">
      <div className="br-100 w1 h1 mr3 bg-green" />
      {children}
    </li>
  );
};

export default UserList;
