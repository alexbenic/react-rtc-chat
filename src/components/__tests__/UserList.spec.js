import React from "react";
import each from "jest-each";
import { render, cleanup } from "react-testing-library";
import randombytes from "randombytes";
import UserList from "../UserList";

const user = name => {
  return {
    id: randombytes(8).toString("hex"),
    nick: name,
    stream: {},
    app: 1
  };
};

afterEach(cleanup);

each([
  ["zvrk", []],
  ["zoom", [user("zoom"), user("maci")]],
  ["maci", [user("zoom"), user("zvrk"), user("zook")]]
]).test("current user %s, list of connected %j", (name, users) => {
  const { container, getByTestId } = render(
    <UserList current={user(name)} users={users} />
  );

  const userList = container.querySelectorAll("li");

  expect(userList.length).toBe(users.length);
});

test("snapshot", () => {
  const { container } = render(
    <UserList current={user("zvrk")} users={[user("maci")]} />
  );

  expect(container.firstChild).toMatchSnapshot();
});
