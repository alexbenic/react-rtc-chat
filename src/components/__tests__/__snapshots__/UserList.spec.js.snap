// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="flex-auto mt2"
>
  <li
    class="flex items-center mt2 mb2 pb1 pt1"
  >
    <div
      class="br-100 w1 h1 mr3 bg-green"
    />
    maci
  </li>
</div>
`;
